File: soln-array-problems/peak-index-in-a-mountain-array.java
/*
https://leetcode.com/problems/peak-index-in-a-mountain-array/
*/

import java.util.*;
import java.lang.*;
import java.io.*;

class Solution {
    public int peakIndexInMountainArray(int[] A) {
       if(A.length<3){
            return -1;
        }

        int i=0;
        while(i<A.length && i+1<A.length && A[i]<A[i+1])
            i++;

        if(i==0 || i+1>=A.length)
            return -1;

        return i;
    }
}

File: C++/soln-binary-search-problems/peak-index-in-a-mountain-array.cpp
/**
 @author Farheen Bano
  
 Date 15-07-2021
 
 Reference-
 https://leetcode.com/problems/peak-index-in-a-mountain-array/
*/

class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int start=0, end=arr.size()-1;
        
        while(start<end){
            int mid=start+(end-start)/2;
            if(arr[mid]>arr[mid+1])
                end=mid;
            else
                start=mid+1;
        }
        return start;
        
    }
};